rpm: drop use of $FIRST_ARG in ceph-immutable-object-cache

smithfarm opened the pull request:
The use of $FIRST_ARG was probably required because the SUSE-specific
%service_* rpm macros were playing tricks on the shell positional parameters.
This is bad practice and error-prone, so let's assume that no macros should do
that anymore and hence it's safe to assume that positional parameters remain
unchanged after any rpm macro call.

Thanks to Franck Bui for providing the original patch
926433f5d45e557c42f050b43798ba29dc495e02 that this patch is modeled after.

NOTE: the use of FIRST_ARG had already been eliminated by
926433f5d45e557c42f050b43798ba29dc495e02 but was re-introduced later by
9466d7098573dafcfede5e9c852374fbbd99f9e7

Fixes: 9466d7098573dafcfede5e9c852374fbbd99f9e7

Fixes: https://tracker.ceph.com/issues/51797

Signed-off-by: Nathan Cutler <ncutler@suse.com>

smithfarm commented:
NOTE TO SELF: backport to pacific

idryomov reviewed (changes requested):
Since Franck is the author, the commit needs to have Franck's sign-off.  Looks like we need to improve the Signed-off-by check...

idryomov commented:
> NOTE TO SELF: backport to pacific

Only to pacific?  Immutable object cache was added in octopus so I'm pretty sure `FIRST_ARG` is there as well.

smithfarm commented:
@idryomov Requested changes made. Thanks!

smithfarm commented:
@idryomov You can rest assured I do have Franck's permission to submit this here on his behalf.

smithfarm commented:
> Since Franck is the author

Franck is the author of the original patch and the original idea of eliminating FIRST_ARG came from him, yes.

I am the author of this patch.

smithfarm commented:
> Looks like we need to improve the Signed-off-by check...

We should be careful with this, lest we shut out contributors without GitHub accounts, who are forced to submit patches through proxies.

idryomov commented:
> > Since Franck is the author
> 
> Franck is the author of the original patch [9466d70](https://github.com/ceph/ceph/commit/9466d7098573dafcfede5e9c852374fbbd99f9e7) and the original idea of eliminating FIRST_ARG came from him, yes.
> 
> I am the author of this patch.

If you are the author of this patch, your name and email should be in the `Author` field in git.  Credits should go in the patch description, either free-form (e.g. "This patch is based on ... from ..." or "This resurrects a patch from ...") or as a tag such as `Suggested-by` (in case you wrote the patch but the idea/inspiration came from someone else and you want to acknowledge them).

idryomov commented:
> We should be careful with this, lest we shut out contributors who are forced to submit patches through proxies because they do not possess a GitHub account.

The requirement is that the author of the patch (the person whose name and email are in the `Author` field) signs the patch with their Signed-off-by (same name and email as in the `Author` field).  If there is another person involved (e.g. because the author doesn't have a Github account or because the patch is being submitted as part of a larger series authored by that other person), that person appends their Signed-off-by indicating that the patch went through them.  It is exactly the same as in projects that do everything via email, so no one should be shut off.  (And we are happy to take Ceph patches via email too, when asked.)

smithfarm commented:
OK, I had Franck as the "git author" but that was an oversight. Corrected!

smithfarm commented:
@idryomov Thanks for the clarification. When you wrote "author" I was not thinking about git's "author" field. I'm the author of this patch in that sense, and in the commit message I am crediting Franck for being the author of the original patch that would have implemented this fix had the code existed at that time.

I don't think Franck is interested in contributing here. He asked me to submit the original patch on his behalf.

My main concern here is to get the fixes in. I less concerned about what the author field says or whom the patch is credited to. Hopefully I have now addressed all the concerns. If not, let me know :-D

idryomov commented:
`AuthorDate` says that you authored this patch over two years ago (Feb 28 10:55:22 2019), but who knows ;)

smithfarm commented:
> `AuthorDate` says that you authored this patch over two years ago (Feb 28 10:55:22 2019), but who knows ;)

Time is relative ;-)

tchaikov commented:
https://pulpito.ceph.com/kchai-2021-07-24_05:22:39-rados-wip-kefu-testing-2021-07-24-1158-distro-basic-smithi/
